pyccl/pk2d.py

```diff
@@ -248,13 +253,145 @@ def eval_dlogpk_dlogk(self, k, a, cosmo):
 
         return f
 
+    def get_spline_arrays(self):
+        """Get the spline data arrays.
+
+        Returns:
+            a_arr: array_like
+                Array of scale factors.
+            lk_arr: array_like
+                Array of logarithm of wavenumber k.
+            pk_arr: array_like
+                Array of the power spectrum P(k, z). The shape
+                is (a_arr.size, lk_arr.size).
+        """
+        if not self.has_psp:
+            raise ValueError("Pk2D object does not have data.")
+
+        a_arr, lk_arr, pk_arr = _get_spline2d_arrays(self.psp.fka)
+        if self.psp.is_log:
+            pk_arr = np.exp(pk_arr)
+
+        return a_arr, lk_arr, pk_arr
+
     def __del__(self):
         """Free memory associated with this Pk2D structure
         """
         if hasattr(self, 'has_psp'):
             if self.has_psp and hasattr(self, 'psp'):
                 lib.f2d_t_free(self.psp)
 
+    def _check_Pk2D_compatability(self, other):
+        if not isinstance(other, Pk2D):
+            raise ValueError("Binary operator of Pk2D objects is only defined "
+                             "for other Pk2D objects.")
+        if not self.has_psp or not other.has_psp:
+            raise ValueError("Pk2D object does not have data.")
+        if (self.psp.lkmin < other.psp.lkmin
+                or self.psp.lkmax > other.psp.lkmax):
+            raise ValueError("The 2nd operand has its data defined over a "
+                             "smaller k range than the 1st operand.")
+        if (self.psp.amin < other.psp.amin
+                or self.psp.amax > other.psp.amax):
+            raise ValueError("The 2nd operand has its data defined over a "
+                             "smaller a range than the 1st operand.")
+
+        a_arr_a, lk_arr_a, pk_arr_a = self.get_spline_arrays()
+        a_arr_b, lk_arr_b, pk_arr_b = other.get_spline_arrays()
+        if not (a_arr_a.size == a_arr_b.size and lk_arr_a.size == lk_arr_b.size
+                and np.allclose(a_arr_a, a_arr_b)
+                and np.allclose(lk_arr_a, lk_arr_b)):
+            # Since the power spectrum is evalulated on a smaller support than
+            # where it was defined, no extrapolation is necessary and the
+            # dependence on the cosmology in moot.
+            # CosmologyVanillaLCDM is being imported here instea of the top of
+            # the module due to circular import issues there.
+            from .core import CosmologyVanillaLCDM
+            dummy_cosmo = CosmologyVanillaLCDM()
+            pk_arr_b = np.array([other.eval(k=np.exp(lk_arr_a),
+                                            a=a_,
+                                            cosmo=dummy_cosmo)
+                                 for a_ in a_arr_a])
+        return a_arr_a, lk_arr_a, pk_arr_a, pk_arr_b
```

Comment:
The function name is `check_compatibility` but it doesn't only check stuff, it returns some compatible arrays.